### PR TITLE
Master support netcore 1 1

### LINKED ISSUE #675: Add support for .NET Core 1.1 to the VS Extension
We need to support deploying .NET core 1.1 projects.

### PR DESCRIPTION
**Pull request by ivannaranjo**
This PR adds support for .NET Core 1.1 apps. This PR also adds support for determining the name of the entrypoint assembly based on the published app, instead of just looking at the project name.

The project name will still be used for the case where the user wants us to generate a Dockerfile, the user can update the Dockerfile as needed should the name of the project change.

Fixes #675

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/ChoiceStep/ChoiceStepViewModel.cs [hunk 1]
```diff
@@ -133,5 +133,8 @@ public static IPublishDialogStep CreateStep()
 
             return viewModel;
         }
+
+        private static bool IsSupportedNetCoreProject(IParsedProject project)
+            => project.ProjectType == KnownProjectTypes.NetCoreWebApplication1_0 || project.ProjectType == KnownProjectTypes.NetCoreWebApplication1_1;
```
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/ChoiceStep/ChoiceStepViewModel.cs]: This line is over 120 characters.

File: GoogleCloudExtension/GoogleCloudExtension.Deployment/CommonUtils.cs [hunk 2]
```diff
@@ -25,33 +26,29 @@ namespace GoogleCloudExtension.Deployment
...
-        /// Returns the project name given the path to the project.json. If the project.json file defines a
-        /// "name" property then it is used as the name for the final assembly, otherwise the name of the directory
-        /// is used as the name of the final assembly.
+        /// Returns the name of the entrypoint assembly for the .NET Core project given it's stage directory. The name
+        /// is determined by looking for the .deps.json that defines the app's structure.
         /// </summary>
-        /// <param name="projectPath">The full path to the project.json of the project.</param>
-        internal static string GetProjectName(string projectPath)
+        /// <param name="stageDirectory">The directory where the app is being staged.</param>
+        /// <returns></returns>
```
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension.Deployment/CommonUtils.cs]: Empty `<returns>`? The comments are complete without it, so you can just remove it.

File: GoogleCloudExtension/GoogleCloudExtension.Deployment/GoogleCloudExtension.Deployment.csproj.user [hunk 3]
```diff
@@ -1,6 +1,6 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (ILMTitan) [inline GoogleCloudExtension/GoogleCloudExtension.Deployment/GoogleCloudExtension.Deployment.csproj.user]: This file should be deleted from the index and added to .gitignore.

File: GoogleCloudExtension/GoogleCloudExtension.Deployment/CommonUtils.cs [hunk 4]
```diff
@@ -25,33 +26,29 @@ namespace GoogleCloudExtension.Deployment
     /// </summary>
     internal static class CommonUtils
     {
+        // This pattern is to be used to find al of  the .deps.json in the stage directory.
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.Deployment/CommonUtils.cs]: find "all" of

File: GoogleCloudExtension/GoogleCloudExtension.Deployment/CommonUtils.cs (hunk 2 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension.Deployment/CommonUtils.cs]: done.

File: GoogleCloudExtension/GoogleCloudExtension.Deployment/CommonUtils.cs (hunk 4 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension.Deployment/CommonUtils.cs]: done.

File: GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/ChoiceStep/ChoiceStepViewModel.cs (hunk 1 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension/PublishDialogSteps/ChoiceStep/ChoiceStepViewModel.cs]: Ordinarily not very sad about having long lines, but I agree that this one is particularly unyielding. Done.

**Pull request merged**